### PR TITLE
[WIP] Continuous-action inverse model

### PR DESCRIPTION
**Pull request by camall3n**

### CODE REVIEW

File: abstractions/common/modules.py [hunk 1]
```diff
@@ -122,19 +122,33 @@ class InverseModel(torch.nn.Module):
...
-        return self.model(context)
+        shared_vector = self.body(context)
+
+        if self.discrete:
+            return self.log_pr_linear(shared_vector)
+        else:
+            mean = self.mean_linear(shared_vector)
+            log_std = self.log_std_linear(shared_vector)
+            log_std = torch.clamp(log_std, min=LOG_SIG_MIN, max=LOG_SIG_MAX)
+            std = log_std.exp()
```
Comment (camall3n) [inline abstractions/common/modules.py:173]: Note that I left std as linear -> clamp -> exp, rather than linear -> softplus, because the former looked like it had a smoother loss plot.

Comment (neevparikh) [reply abstractions/common/modules.py:173]: Reasonable.

File: abstractions/common/modules.py [hunk 2]
```diff
@@ -171,25 +185,36 @@ class MarkovHead(torch.nn.Module):
...
             The number of actions in the environment
-    
+
     Notes:
         - This does not support continuous action spaces right now. TODO
     """
     def __init__(self, args, feature_size, num_actions):
         super(MarkovHead, self).__init__()
-        self.inverse_model = InverseModel(args, feature_size, num_actions)
+        self.discrete = hasattr(args, 'discrete') and args.discrete
```
Comment (neevparikh) [inline abstractions/common/modules.py:217]: I'm not sure we want this to be False in the default case no? Maybe a raise ValueError is acceptable here.

**Review by neevparikh — COMMENTED** [review]
Overall, think it's good! I have a small comment about the discrete attribute default, check the other comments. I think we can merge after we add the SAC + MA option in.

File: abstractions/common/modules.py (hunk 2 above)
Comment (camall3n) [reply abstractions/common/modules.py:217]: I don't like using `hasattr()` here, but at the moment the `discrete` arg only exists in the sac_parser. I think a better solution would pull that arg into the main parser, and the ideal solution would be to automatically determine whether the action space is continuous by introspecting the environment
